src/remediations/remediations.queries.js

```diff
@@ -169,6 +183,75 @@ exports.list = function (
             }
         }
 
+        // status filter
+        if (filter.status) {
+            const statusFilter = {};
+            const fields = { fields: { data: ['labels', 'status'] } };
+
+            // get all playbook runs from dispatcher
+            const dispatcherRuns = await dispatcher_impl.fetchPlaybookRuns(statusFilter, fields);
+
+            // map playbook-run id to status
+            const runStatusById = {};
+            for (const run of dispatcherRuns.data) {
+                const runId = run.labels?.['playbook-run'];
+                if (runId) {
+                    runStatusById[runId] = run.status;
+                }
+            }
+
+            // Map playbook-run id to remediation_id
+            const playbookRunsForRemediation = await db.playbook_runs.findAll({
+                attributes: ['id', 'remediation_id'],
+                where: {
+                    id: {
+                        [Op.in]: Object.keys(runStatusById)
+                    }
+                },
+                raw: true
+            });
+
+            // We need to get the count of playbook run statuses per remediation
+            // This will be used to compute the overall(aggregate) remediation status later
+            const remediationStatusMap = {};
+            for (const {id, remediation_id} of playbookRunsForRemediation) {
+                // Initialize count object for this remediation if not already done
+                if (!remediationStatusMap[remediation_id]) {
+                    remediationStatusMap[remediation_id] = {
+                        count_success: 0,
+                        count_failure: 0,
+                        count_timeout: 0,
+                        count_running: 0
+                    };
+                }
+
+                // Increment the appropriate count for this remediation based on the run's status
+                const status = runStatusById[id];
+                const counts = remediationStatusMap[remediation_id];
+
+                if (status === 'success') {
+                    counts.count_success++;
+                } else if (status === 'failure') {
+                    counts.count_failure++;
+                } else if (status === 'timeout') {
+                    counts.count_timeout++;
+                } else if (status === 'running') {
+                    counts.count_running++;
+                }
+            }
```

Comment:
This gets a little complicated but I don't think there's really a better way of doing it... It's just a complicated thing that we need to do :/